src/System.Private.CoreLib/src/System/Threading/ClrThreadPoolPreAllocatedOverlapped.cs

```diff
@@ -48,7 +49,7 @@ public sealed class PreAllocatedOverlapped : IDisposable, IDeferredDisposable
         ///     This method was called after the <see cref="ThreadPoolBoundHandle"/> was disposed.
         /// </exception>
         [CLSCompliant(false)]
-        public PreAllocatedOverlapped(IOCompletionCallback callback, object state, object pinData)
+        public PreAllocatedOverlapped(IOCompletionCallback callback, object? state, object? pinData)
```

Comment:
Github won't let me comment on that line, but should `_overlapped` be marked as nullable? It seems we rely on it being null to see whether we free it or not in `IDeferredDisposable.OnFinalRelease` -- since it is internal it doesn't matter much I guess, but it seems more honest.

Reply:
I believe the only way it could be null is if the ctor threw an exception, such that the object never got fully initialized, and then it got finalized.  Otherwise, it's readonly and is initialized in the ctor.  I'll add a comment to the OnFinalRelease, which may be called from the finalizer, but I don't think the field needs to be made nullable... otherwise pretty much every field on a type where the ctor can throw would need to be nullable.

Reply:
Ok, makes sense -- thanks for elaborating.